Chap_API_Sync_Access.tex

```diff
@@ -771,6 +782,10 @@ \subsection{Query attributes}
 \declareAttributeNEW{PMIX_SERVER_INFO_ARRAY}{"pmix.srv.arr"}{pmix_data_array_t}{
 Array of \refstruct{pmix_info_t} about a given server, starting with its \refattr{PMIX_NSPACE} and including at least one of the rendezvous-required pieces of information.
 }
+%
+\declareAttributeNEW{PMIX_QUERY_ABI_VERSION}{"pmix.qry.abiversion"}{char *}{
+Query the \ac{ABI} version supported by the \ac{PMIx} library. The version returned will be of the form \code{"MAJOR.MINOR"}. For example. \code{"4.2"} representing version v4.2 of the \ac{PMIx} \ac{ABI}. NO QUALIFIERS.
+}
```

Comment:
Notes from April 14, 2022 meeting
 * Good discussion about how the "Major.Minor" will be updated and how an application might use this information.
 * What is the guidance on updating the "Major.Minor"?
   * If a new API is added, this does not break the ABI but does change the ABI. So increment the "Minor".
   * If an API is removed, this does break the ABI. So increment the "Major" and set "Minor" to 0.
   * If an API is modified, this does break the ABI. So increment the "Major" and set "Minor" to 0.
 * If an application builds against "`1.7`" of the ABI.
   * They require a major version of "`1`" and a minor version of "`x >= 7`" since they do not care about additions to the ABI only that the `1.7` ABI is still intact.
   * If the major version is not "`1`" then they need to consult the ABi to determine if the modifications/removals impact their application.
 * The complexity comes with Provisional items that are allowed to change form between minor revisions. This is in contrast to Stable items that have a longer Deprecation/Removal process.
   * If a Provisional API is changed then it changes the "Major" number in the ABI version causing broad disruption.
   * The suggestion was to separate the Stable and Provisional interfaces (as best as possible) and have two queries. `PMIX_QUERY_ABI_VERSION_STABLE` and `PMIX_QUERY_ABI_VERSION_PROVISIONAL`. This will help isolate applications focused on stable interfaces from rapid development of provisional items.
 * In some circumstances an implementation may continue to provide deprecated/removed interfaces. In this case they support multiple versions of the ABI. Consider two ABI versions `1.7` and `2.0` where the only difference is the removal of `PMIx_Foo()`
   * They support the old version `1.7` since it still includes `PMIx_Foo()`.
   * They also support the new version `2.0`. They just have an additional function `PMIx_Foo()` that is not defined in the `2.0` ABI.
   * As such it is important to allow an implementation to report more than one ABI level.
   * Suggestion: allow for a comma separated list to be returned.
   * If the difference between `1.7` and `2.0` is a modification of an existing API/structure then an implementation must determine which ABI they conform to. The implementation might choose to add a configure time option to choose the ABI level. Other runtime level options are available though more complex to implement.

Reply:
FYI. I added this to the Governance language
 * https://github.com/pmix/governance/pull/34/files#r851349685
I'm going to suggest that we also add it to the PMIx Standard (in PR #375) then reference that section here

Reply:
`pmix-abi` ticket to identify the Stable vs Provisional ABI:
 * https://github.com/pmix/pmix-abi/issues/11

Reply:
I pushed a new commit to address this comment.